src/uct/ib/base/ib_iface.c

```diff
@@ -1011,6 +1024,9 @@ UCS_CLASS_INIT_FUNC(uct_ib_iface_t, uct_ib_iface_ops_t *ops, uct_md_h md,
         self->is_global_addr = 0;
     }
 
+    self->config.traffic_class      = config->traffic_class;
```

Comment:
you already set it on line [946](https://github.com/openucx/ucx/pull/3061/files#diff-694c9dba586b3d5c9d63a14795983496R946)